mlir/lib/Dialect/MIOpen/MIOpenDialect.cpp

```diff
@@ -648,17 +649,19 @@ ParseResult TransformingForOp::parse(OpAsmParser &parser,
 
   llvm::SMLoc iterArgsLoc = parser.getCurrentLocation();
   llvm::SmallVector<OpAsmParser::Argument> iterArgs;
-  llvm::SmallVector<OperandType> iterInits;
+  llvm::SmallVector<OpAsmParser::UnresolvedOperand> iterInits;
   llvm::SmallVector<mlir::Type> iterTypes;
   if (parser.parseOptionalKeyword("iter_args").succeeded()) {
-    if (parseAssignmentListWithTypes(iterArgs, iterInits, iterTypes, parser)) {
+    if (parser.parseAssignmentList(iterArgs, iterInits) ||
+        parser.parseArrowTypeList(iterTypes)) {
```

Comment:
Remove the definition of parseAssignmentListWithTypes(). It's no longer needed.

Reply:
Done, good catch